sdk/core/Azure.Core/src/DynamicData/MutableJsonElement.cs

```diff
@@ -795,13 +786,8 @@ public void RemoveProperty(string name)
                 throw new InvalidOperationException($"Object does not have property: '{name}'.");
             }
 
-            Dictionary<string, object> dict = JsonSerializer.Deserialize<Dictionary<string, object>>(GetRawBytes(), _root.SerializerOptions)!;
```

Comment:
Same comment as https://github.com/Azure/azure-sdk-for-net/pull/36791/files#r1228357370